### PR TITLE
EL-4251 - Menu: using the same submenu for multiple menu items doesn't work

### PR DESCRIPTION
**Pull request by JonathanToddMF**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue

https://portal.digitalsafe.net/browse/EL-4251

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4251-menu-submenu-multiple-items

#### Downstream Build(s)

* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4251-menu-submenu-multiple-items~CI/)

### CODE REVIEW

**Comment by JonathanToddMF** [comment]
Hi @donna-marijne and @ashley-hunter 

I have got a little stuck on how to fix the remaining bug in my fix for this issue. 

The following plunker shows my current implementation - https://plnkr.co/edit/PRkvsj0aD5lJeJTH

The problem I am facing is when your mouse leaves the dropdown menu and moves down across 2 menu items. The menu doesn't seem to open. I believe the issue lies with the _menuShouldCLose getter in the menu-trigger.directive.ts. It seems something here is making the menu close when it shouldn't.

Here is a video detailing the problem - https://gyazo.com/42c5f5218a45aa9553655a170f2cc4be

I have trying to emulate the code using in angular material here - https://github.com/angular/components/blob/53d2d4bd084d98017be9ab8dde9f1db6f3339a52/src/material/menu/menu-trigger.ts#L555-L585

Thank you,
Jonathan

**Comment by JonathanToddMF** [comment]
Ignore the above, my original solution worked, just needed await in the unit test!

File: src/components/menu/menu-trigger/menu-trigger.directive.ts [hunk 1]
```diff
@@ -236,6 +236,14 @@ export class MenuTriggerDirective implements OnInit, OnDestroy {
         }
     }
 
+    @HostListener('mousemove')
+    _onMouseMove(): void {
+        if (this._isSubmenuTrigger && !this._parentMenu._isAnimating) {
+            timer(this._debounceTime).pipe(takeUntil(this._onDestroy$))
```
Comment (donna-marijne) [inline src/components/menu/menu-trigger/menu-trigger.directive.ts:242]: Does `setTimeout` work here?

Comment (JonathanToddMF) [reply src/components/menu/menu-trigger/menu-trigger.directive.ts:242]: It does indeed, I will push up the change now

**Review by donna-marijne — DISMISSED** [review]
Verification:

https://plnkr.co/edit/C3ObaxIz9edAanUM (open source) - :heavy_check_mark:
https://plnkr.co/edit/So7IGB04v6sWFh7O (Micro Focus) - :heavy_check_mark:

**Review by Roland-Penny — APPROVED** [review]
Accepted

**Comment by JonathanToddMF** [comment]
Merge complete and build passing, just need to wait on the ux-aspects-micro-focus build - https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4251-menu-submenu-multiple-items~CI/

**Comment by JonathanToddMF** [comment]
both builds passing now, this is ready to merge

**Review by donna-marijne — APPROVED** [review]

**Pull request merged**